src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -3425,6 +3428,70 @@ void ride_construction_toolupdate_construct(sint32 screenX, sint32 screenY)
             _currentTrackBeginZ += 16;
     }
 
+    if (!_shopManualRotation &&
+        _rideConstructionState == RIDE_CONSTRUCTION_STATE_PLACE &&
+        ride_type_has_flag(ride->type, RIDE_TYPE_FLAG_IS_SHOP))
+    {
+        rct_tile_element *pathsByDir[4];
+        constexpr std::pair<sint8, sint8> DirOffsets[4] = {{-1, 0}, {0, 1}, {1, 0}, {0, -1}};
```

Comment:
No, that's improper use of LocationXY8, which takes Tile Coordinates. std::pair is fine.

Reply:
Yes, but this is `sLocationXY8` (with the `s` prefix), which used signed bytes and can't store the coordinates for every possible tile. `DirOffsets[i].x` and `.y` rather than `DirOffsets[i].first` and `.second` is also clearer in my opinion.

Reply:
Right, ok. Let's go with your suggestion then.